django/core/serializers/xml_serializer.py

```diff
@@ -387,6 +394,8 @@ def m2m_convert(n):
         try:
             for c in node.getElementsByTagName("object"):
                 values.append(m2m_convert(c))
+        except UnexpectedNode:
+            raise
```

Comment:
There might be another one of these to filter out and raise, e.g. in `save_deferred_fields()`.